### PR TITLE
Create GH Action codestart

### PR DESCRIPTION
**Pull request by ia3andy**

### CODE REVIEW

**Comment by ia3andy** [comment]
CI failure is unrelated (neo4j)

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/gradle/codestart.yml [hunk 1]
```diff
@@ -19,9 +19,10 @@ language:
         cmd:
           dev: gradle quarkusDev
           package: gradle quarkusBuild
-          package-uberjar:  gradle quarkusBuild --uber-jar
+          package-uberjar: gradle quarkusBuild --uber-jar
           package-native: gradle build -Dquarkus.package.type=native
           package-native-container: gradle build -Dquarkus.package.type=native -Dquarkus.native.container-build=true
+          build-ci: gradle build
```
Comment (maxandersen) [inline devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/gradle/codestart.yml:25]: this shuold be `./gradlew build` to ensure it uses the right gradle

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml [hunk 2]
```diff
@@ -25,6 +25,7 @@ language:
         cmd:
           dev: mvn compile quarkus:dev
           package: mvn package
-          package-uberjar:  mvn package -PuberJar
```
Comment (maxandersen) [inline devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml:28]: shouldn't all the refereences to mvn be `./mvn`

Comment (maxandersen) [reply devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml:28]: ooh this is when no mvn wrapper....you need to check if the setup java gives you the right maven/gradle version

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/gradle/codestart.yml (hunk 1 above)
Comment (maxandersen) [reply devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/gradle/codestart.yml:25]: same as below - need to check we get the right/new enough version of gradle in the build

Comment (ia3andy) [reply devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/gradle/codestart.yml:25]: I will fix that in another iteration, I need this to make master work with code.quarkus

Comment (ia3andy) [reply devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/gradle/codestart.yml:25]: https://github.com/quarkusio/quarkus/issues/12438

**Review by maxandersen — APPROVED** [review]
lgtm

**Pull request merged**